### PR TITLE
Assets manager force download and success checks

### PR DESCRIPTION
**Pull request by victorbenichoux**
Here I implement two additional systems in `modelkit` to ensure assets are correctly downloaded.

First: a globale environment variable to force the download of all assets. when `MODELKIT_STORAGE_FORCE_DOWNLOAD` is set to `True`, the `AssetsManager` will re-download all assets, even though they are already present in the `ASSETS_DIR`.

Second: when an asset is downloaded, the AssetsManager will also touch a `.SUCCESS` file (in directory assets), or a `name+.SUCCESS` file (in file assets). Whenever it looks for an asset, if this file is not found it will invalidate the local files and re-download.

### CODE REVIEW

File: modelkit/assets/settings.py [hunk 1]
```diff
@@ -59,6 +62,12 @@ def dispatch_settings(cls, fields):
 
         return fields
 
+    @validator("storage_force_download", pre=True)
+    def force_download(cls, v):
+        if v == "True" or v is True:
+            return True
+        return False
```
Comment (antoinejeannot) [inline modelkit/assets/settings.py:69]:
```suggestion
        return v == "True" or v is True:

```

File: modelkit/utils/cache.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import abc
 import hashlib
 import pickle
+from dataclasses import dataclass
```
Comment (antoinejeannot) [inline modelkit/utils/cache.py:4]: Weird that linting passed before

File: modelkit/assets/settings.py (hunk 1 above)
Comment (antoinejeannot) [reply modelkit/assets/settings.py:69]: NB: ^

**Review by antoinejeannot — APPROVED** [review]
Great work as always !

File: modelkit/utils/cache.py (hunk 2 above)
Comment (victorbenichoux) [reply modelkit/utils/cache.py:4]: Indeed, I am not so sure why...

**Review by antoinejeannot — APPROVED** [review]

**Pull request merged**